app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.kt

```diff
@@ -140,170 +153,201 @@ abstract class SFragment : Fragment(), Injectable {
-        }
-        val menu = popup.menu
-        val openAsItem = menu.findItem(R.id.status_open_as)
-        val openAsText = (activity as BaseActivity?)?.openAsText
-        if (openAsText == null) {
-            openAsItem.isVisible = false
-        } else {
-            openAsItem.title = openAsText
-        }
-        val muteConversationItem = menu.findItem(R.id.status_mute_conversation)
-        val mutable = statusIsByCurrentUser || accountIsInMentions(activeAccount, status.mentions)
-        muteConversationItem.isVisible = mutable
-        if (mutable) {
-            muteConversationItem.setTitle(
-                if (status.muted != true) {
-                    R.string.action_mute_conversation
-                } else {
-                    R.string.action_unmute_conversation
-                }
-            )
-        }
-        popup.setOnMenuItemClickListener { item: MenuItem ->
-            when (item.itemId) {
-                R.id.post_share_content -> {
-                    val statusToShare = status.reblog ?: status
-                    val sendIntent = Intent().apply {
-                        action = Intent.ACTION_SEND
-                        type = "text/plain"
-                        putExtra(
-                            Intent.EXTRA_TEXT,
-                            "${statusToShare.account.username} - ${statusToShare.content.parseAsMastodonHtml()}"
-                        )
-                        putExtra(Intent.EXTRA_SUBJECT, statusUrl)
+            val menu = popup.menu
+            val openAsItem = menu.findItem(R.id.status_open_as)
+            val openAsText = (activity as BaseActivity?)?.openAsText
+            if (openAsText == null) {
+                openAsItem.isVisible = false
+            } else {
+                openAsItem.title = openAsText
+            }
+            val muteConversationItem = menu.findItem(R.id.status_mute_conversation)
+            val mutable =
+                statusIsByCurrentUser || accountIsInMentions(activeAccount, status.mentions)
+            muteConversationItem.isVisible = mutable
+            if (mutable) {
+                muteConversationItem.setTitle(
+                    if (status.muted != true) {
+                        R.string.action_mute_conversation
+                    } else {
+                        R.string.action_unmute_conversation
                     }
-                    startActivity(
-                        Intent.createChooser(
-                            sendIntent,
-                            resources.getText(R.string.send_post_content_to)
+                )
+            }
+
+            val translateItem = menu.findItem(R.id.status_translate)
+            translateItem.isVisible =
+                onMoreTranslate != null && status.language != Locale.getDefault().language && instanceInfoRepository.getCachedInstanceInfoOrFallback().translationEnabled == true
```

Comment:
I fetch it here before showing the dropdown but I could also preload it when the screen is opened

Reply:
I'd preload it, otherwise there might be an annoying delay when clicking the more button

Reply:
doesn't seem to be a problem a practice but I can preload!